src/vec-push-pop.md

```diff
@@ -25,21 +55,36 @@ pub fn push(&mut self, elem: T) {
         ptr::write(self.ptr.offset(self.len as isize), elem);
     }
 
-    // Can't fail, we'll OOM first.
+    // 絶対成功します。 OOM はこの前に起こるからです。
     self.len += 1;
 }
 ```
 
+<!--
 Easy! How about `pop`? Although this time the index we want to access is
 initialized, Rust won't just let us dereference the location of memory to move
 the value out, because that would leave the memory uninitialized! For this we
 need `ptr::read`, which just copies out the bits from the target address and
 interprets it as a value of type T. This will leave the memory at this address
 logically uninitialized, even though there is in fact a perfectly good instance
 of T there.
+-->
+
+簡単です! では `pop` はどうでしょうか? この場合、アクセスしたいインデックスにある
+値は初期化されていますが、 Rust はメモリ上の値をムーブするために、その場所への
+参照外しをする事を許可しません。なぜならこれによって、メモリを未初期化のままにするからです!
+これに関しては、 `ptr::read` を必要とします。これは、単にターゲットのアドレスから
+ビットをコピーし、それを型 `T` の値として解釈します。これによって、実際には
+そのアドレスのメモリにある値は完全に T のインスタンスであるけれども、
+値を論理的には未初期化の状態のままにします。
```

Comment:
I don't understand the original text of this paragraph.  The translation looks great anyway.